src/main/webapp/js/components/csv/csv-config-create.jsx

```diff
@@ -63,20 +83,32 @@ class CsvConfigCreate extends Component {
           isNew
           onSubmit={this.onSubmit}
           entityType={this.state.entity}
-          csvConfig={{ entity: this.state.entity }}
+          csvConfig={{
+            entity: this.state.entity,
+            csvFields: this.state.initialValues,
+          }}
         />
       </div>
     );
   }
 }
 
+const mapStateToProps = state => ({
+  currentState: state,
```

Comment:
You don't need the entire redux state here, just the field configs, you must remember that when you map the part of redux state any changes to that part will trigger the component update